src/coreclr/src/vm/stackwalk.cpp

```diff
@@ -2266,7 +2266,6 @@ StackWalkAction StackFrameIterator::NextRaw(void)
         // make sure we're not skipping a different transition
         if (m_crawl.pFrame->NeedsUpdateRegDisplay())
         {
-            CONSISTENCY_CHECK(m_crawl.pFrame->IsTransitionToNativeFrame());
```

Comment:
I hit a case where we were walking a frame which had `NeedsUpdateRegDisplay()`, but not `IsTransitionToNativeFrame()`.  I did a quick audit of the frames and this case was clearly valid.  There are frames which have this combination, so it appeared the consistency check was bogus.

Reply:
I would like to see the case. This should not be influenced by your changes and we've never seen this assert to fire before.

Reply:
It is possible this can be simply reverted.  It occurred in early debugging when I was trying to use a SPC.dll w/o crossgen (which was probably also mismatched).

Reply:
Reverted